packages/botonic-react/src/webchat/webchat.jsx

```diff
@@ -486,7 +486,11 @@ export const Webchat = forwardRef((props, ref) => {
 
   useEffect(() => {
     if (webchatState.isWebchatOpen && props.onOpen) props.onOpen()
-    if (!webchatState.isWebchatOpen && props.onClose) props.onClose()
+    if (!webchatState.isWebchatOpen && props.onClose) {
+      props.onClose()
+      setEmojiPickerIsOpened(false)
+      setPersistentMenuIsOpened(false)
```

Comment:
I think that now is a good moment to move these kind of functions into their corresponding actions and dispatchers in order to control these behaviors in an easier way, like the ones that we have in  `hooks.js`'s `useWebchat()`. I think it should be quite straight-forward. Let me know if you need a helping hand on this 💪

Reply:
I don't follow the technicalities, but splitting the code sounds cool :+1: